Tribler/Main/metadata-injector.py

```diff
@@ -70,154 +87,87 @@ def initialize(self):
         self.session.add_observer(self.dispersy_started, NTFY_DISPERSY, [NTFY_STARTED])
         self.session.start()
 
-    def init_managers(self):
-        self._logger.info(u"Initializing managers...")
-        torrent_manager = TorrentManager(None)
-        library_manager = LibraryManager(None)
-        channel_manager = ChannelManager()
-        torrent_manager.connect(self.session, library_manager, channel_manager)
-        library_manager.connect(self.session, torrent_manager, channel_manager)
-        channel_manager.connect(self.session, library_manager, torrent_manager)
+    def shutdown(self):
+        self.cancel_all_pending_tasks()
 
-        #torrent_state_manager = TorrentStateManager()
-        #torrent_state_manager.connect(torrent_manager, library_manager, channel_manager)
+        self._logger.info(u"Shutdown Session...")
+        self.session.shutdown()
+        self.session = None
 
-        self._torrent_manager = torrent_manager
-        self._library_manager = library_manager
-        self._channel_manager = channel_manager
-        #self._torrent_state_manager = torrent_state_manager
+        self._logger.info(u"Sleep for 10 seconds...")
+        time.sleep(10)
 
-    def shutdown(self):
-        self._logger.info(u"Shutting down metadata-injector...")
-        torrentfeed = RssParser.getInstance()
-        torrentfeed.shutdown()
+    @call_on_reactor_thread
+    def dispersy_started(self, *args):
+        # FIXME: because Channel and PreviewChannel are loaded in tribler_main, we need to move them into
```

Comment:
No, LMC is tribler-lib. It provides all the components which are used in Tribler.
Tribler itself is the gui + the communities which are enabled.